### PR TITLE
Split hqwebapps/js/widgets based on select2 version

### PR DESCRIPTION
**Pull request by orangejenny**
[select2 needs upgrading](https://github.com/dimagi/commcare-hq/pull/21756#issuecomment-419252990). We have several widgets based on select2 (.ko-select2, .hqwebapp-autocomplete, hqwebapp/js/select2_handler.js, ...) that each need to be upgraded. It's difficult to migrate these widgets one at a time, because some pages depend on multiple widgets (a single page can't use multiple versions of select2 due to namespace and CSS issues).

This PR splits widgets.js into two versions: one that uses select2 v3 and one that uses select2 v4. The select2 calls in widgets.js are pretty basic, so they're already compatible with both v3 and v4 - which version a page uses depends on what *other* select2 dependencies the page has. The effect of this PR is just to make it explicit which select2 version is being used, both for readability and so that r.js can generate an accurate dependency tree.

@millerdev / @pr33thi

### CODE REVIEW

**Comment by orangejenny** [comment]
Rebased and updated description to reflect that the legacy version is v3, not v2. All the 2s in "select2" got me confused.

File: corehq/apps/reports/views.py [hunk 1]
```diff
@@ -194,7 +194,7 @@
 from corehq.apps.hqcase.utils import resave_case
 from corehq.apps.hqwebapp.decorators import (
     use_jquery_ui,
-    use_select2,
+    use_select2_v4,
```
Comment (millerdev) [inline corehq/apps/reports/views.py]: Doesn't look like this is defined in this PR. Did it exist before?

**Review by millerdev — APPROVED** [review]
Comment may be a blocker, not sure.

Creative use of modules to select the version (no pun intended)—I like it.

File: corehq/apps/reports/views.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/reports/views.py]: yes: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/decorators.py#L23

**Pull request merged**